sdk/iot/pnp/tests/cmocka/test_az_iot_pnp_client.c

```diff
@@ -0,0 +1,284 @@
+  assert_precondition_checked(az_iot_pnp_client_init(
+      &client, AZ_SPAN_NULL, test_device_id, test_root_interface_name, NULL));
+}
+
+static void test_az_iot_pnp_client_init_empty_device_id_fails(void** state)
+{
+  (void)state;
+
+  az_iot_pnp_client client;
+  assert_precondition_checked(az_iot_pnp_client_init(
+      &client, test_device_hostname, AZ_SPAN_NULL, test_root_interface_name, NULL));
+}
+
+static void test_az_iot_pnp_client_init_empty_root_interface_name_fails(void** state)
+{
+  (void)state;
+
+  az_iot_pnp_client client;
+  assert_precondition_checked(
+      az_iot_pnp_client_init(&client, test_device_hostname, test_device_id, AZ_SPAN_NULL, NULL));
+}
+
+#endif // NO_PRECONDITION_CHECKING
+
+static void test_az_iot_pnp_client_get_default_options_succeed(void** state)
+{
+  (void)state;
+
+  az_iot_pnp_client_options options = az_iot_pnp_client_options_default();
+  assert_true(az_span_is_content_equal(options.user_agent, AZ_SPAN_NULL));
+  assert_true(az_span_is_content_equal(options.content_encoding, AZ_SPAN_NULL));
+  assert_true(az_span_is_content_equal(options.content_type, AZ_SPAN_NULL));
+}
+
+static void test_az_iot_pnp_client_init_succeed(void** state)
+{
+  (void)state;
+
+  az_iot_pnp_client client;
+  assert_int_equal(
+      az_iot_pnp_client_init(
+          &client, test_device_hostname, test_device_id, test_root_interface_name, NULL),
+      AZ_OK);
+
+  assert_non_null(az_span_ptr(client._internal.iot_hub_client._internal.iot_hub_hostname));
+  assert_string_equal(TEST_ROOT_INTERFACE_NAME, az_span_ptr(client._internal.root_interface_name));
+}
+
+static void test_az_iot_pnp_client_init_custom_options_succeed(void** state)
+{
+  (void)state;
+
+  az_iot_pnp_client_options options = az_iot_pnp_client_options_default();
+  options.user_agent = test_user_agent;
+  options.content_type = test_content_type;
+  options.content_encoding = test_content_encoding;
+
+  az_iot_pnp_client client;
+  assert_int_equal(
+      az_iot_pnp_client_init(
+          &client, test_device_hostname, test_device_id, test_root_interface_name, &options),
+      AZ_OK);
+
+  assert_non_null(az_span_ptr(client._internal.iot_hub_client._internal.iot_hub_hostname));
+  assert_string_equal(
+      TEST_USER_AGENT, az_span_ptr(client._internal.iot_hub_client._internal.options.user_agent));
+  assert_string_equal(TEST_ROOT_INTERFACE_NAME, az_span_ptr(client._internal.root_interface_name));
+  assert_string_equal(TEST_CONTENT_TYPE, az_span_ptr(client._internal.options.content_type));
+  assert_string_equal(
+      TEST_CONTENT_ENCODING, az_span_ptr(client._internal.options.content_encoding));
+}
+
+static void test_az_iot_pnp_client_get_user_name_succeed(void** state)
+{
+  (void)state;
+
+  az_iot_pnp_client client;
+  assert_int_equal(
+      az_iot_pnp_client_init(
+          &client, test_device_hostname, test_device_id, test_root_interface_name, NULL),
+      AZ_OK);
+
+  uint8_t test_span_buffer[sizeof(test_correct_pnp_user_name) - 1];
```

Comment:
just for consistency all of the `sizeof()` calls in this file can probably be switched to `_az_COUNTOF()`.
[Macro found here](https://github.com/Azure/azure-sdk-for-c/blob/35629203f7dab143a65e55760f10ef211f204e1d/sdk/core/core/inc/_az_cfg.h#L92)

Reply:
Fixing.